controller/CosmeticAPIController.py

```diff
@@ -0,0 +1,254 @@
+        product_result = ProductRepo.getProduct(product_id)
+        if product_result is None:
+            return make_response(jsonify(response=f'Product with id {product_id} not found'), 404)
+        if request.method == 'DELETE':
+            ProductRepo.delete(product_result)
+        return jsonify(response=f'Product with id : {product_id} is deleted successfully')
+    except sqlalchemy.exc.InvalidRequestError as error:
+        return make_response(jsonify(error_message=str(error)), 400)
+    except exc.SQLAlchemyError as sqlexp:
+        return make_response(jsonify(error_message=str(sqlexp)), 400)
+    except Exception as excp:
+        return make_response(jsonify(error_message=str(excp)), 400)
+
+
+"""
+API Call to get the brand 
+Input: brand_id 
+Output: Returns the Brand details for the given brand_id
+"""
+
+
+@app.route('/brand/<int:brand_id>', methods=['GET'])
+@jwt_required
+def get_brand_by_id(brand_id):
+    logging.info(f'getBrandById: User has given input brand_id as: {brand_id}')
+    brand = BrandRepo.get(brand_id)
+    if brand is None:
+        return make_response(jsonify(response=f'Brand with id {brand_id} not found'), 404)
+    return jsonify(brand=Brand.serialize(brand))
+
+
+"""
+API Call to get all brands 
+"""
+
+
+@app.route('/getallbrands', methods=['GET'])
+@jwt_required
+def get_all_brands():
+    """
+                @return: Returns the list of JSON Objects i.e product brands
+    """
+    brand = BrandRepo.get_all_brands()
+    if brand is None or len(brand) == 0:
+        return make_response(jsonify(response=f'No Brands Available'), 404)
+    return jsonify(brands=Brand.serialize_list(brand))
+
+
+"""
+API Call to get all categories
+  
+"""
+
+
+@app.route('/getallcategories', methods=['GET'])
+@jwt_required
+def get_all_categories():
+    """
+            @return: Returns the list of JSON Objects i.e product categories
+    """
+    categories = CategoryRepo.get_all_categories()
+    if categories is None or len(categories) == 0:
+        return make_response(jsonify(response=f'No Categories Available'), 404)
+    return jsonify(categories=Category.serialize_list(categories))
+
+
+"""
+API Call to get all colors 
+"""
+
+
+@app.route('/getallcolors', methods=['GET'])
+@jwt_required
+def get_all_colors():
+    """
+        @return: Returns the list of JSON Object i.e product colors
+    """
+    colors = ColorRepo.get_all_colors()
+    if colors is None or len(colors) == 0:
+        return make_response(jsonify(response=f'No Colors Available'), 404)
+    return jsonify(colors=Color.serialize_list(colors))
+
+
+"""
+API Call to get all product types 
+"""
+
+
+@app.route('/getallproducttypes', methods=['GET'])
+@jwt_required
+def get_all_producttypes():
+    """
+    @return: Returns the list of JSON Objects i.e product types
+    """
+    producttypes = ProductTypeRepo.get_all_producttypes()
+    if producttypes is None or len(producttypes) == 0:
```

Comment:
I wonder if this is logically what we really want. What I mean by that is whether we want to return a `404` if there is no product type or if we rather want to return an empty list.